google-cloud-firestore/lib/google/cloud/firestore/database.rb

```diff
@@ -0,0 +1,990 @@
+          doc_path = "#{path}/documents/#{document_path}"
+
+          Document.from_path doc_path, self
+        end
+        alias_method :document, :doc
+
+        ##
+        # Retrieves a list of documents with data.
+        #
+        # @param [String, Document::Reference] docs One or more strings
+        #   representing the path of the document, or document reference
+        #   objects.
+        # @param [Array<String|Symbol>, String|Symbol] mask A list of field
+        #   paths to filter the returned document data by.
+        #
+        # @yield [documents] The block for accessing the documents with data.
+        # @yieldparam [Document::Snapshot] document A document with data.
+        #
+        # @return [Enumerator<Document::Snapshot>] documents with data list.
+        #
+        # @example
+        #   require "google/cloud/firestore"
+        #
+        #   firestore = Google::Cloud::Firestore.new
+        #
+        #   # Get and print city documents
+        #   cities = ["cities/NYC", "cities/SF", "cities/LA"]
+        #   firestore.get_all(cities).each do |city|
+        #     puts "#{city.document_id} has #{city[:population]} residents."
+        #   end
+        #
+        def get_all *docs, mask: nil
+          ensure_service!
+
+          return enum_for(:get_all, docs, mask: mask) unless block_given?
+
+          doc_paths = Array(docs).flatten.map do |doc_path|
+            coalesce_doc_path_argument doc_path
+          end
+
+          results = service.get_documents doc_paths, mask: mask
+          results.each do |result|
+            next if result.result.nil?
+            yield Document.from_batch_result(result, self)
+          end
+        end
+        alias_method :get_docs, :get_all
+        alias_method :get_documents, :get_all
+        alias_method :find, :get_all
+
+        ##
+        # Create a batch to perform multiple changes at the same time.
+        #
+        # All changes are accumulated in memory until the block completes.
+        # Unlike `transaction`, batches are not automatically retried. See
+        # {Batch}.
+        #
+        # @return [Time] The time the changes were committed
+        #
+        # @example
+        #   require "google/cloud/firestore"
+        #
+        #   firestore = Google::Cloud::Firestore.new
+        #
+        #   firestore.batch do |b|
+        #     # Set the data for NYC
+        #     b.col("cities").doc("NYC").set({ name: "New York City" })
+        #
+        #     # Update the population for SF
+        #     b.col("cities").doc("SF").update({ population: 1000000 })
+        #
+        #     # Delete LA
+        #     b.col("cities").doc("LA").delete
+        #   end
+        #
+        def batch
+          batch = Batch.from_database self
+          yield batch
+          batch.commit
+        end
+
+        ##
+        # Create a document with the provided object values.
+        #
+        # The batch will fail if the document already exists.
```

Comment:
It might help to explain why a batch is being used, or simply replace "batch" with "operation" if it can be considered an implementation detail.

Reply:
Changed to "operation".